template/lambda-sqs-worker-cdk/infra/index.ts

```diff
@@ -15,3 +16,11 @@ new AppStack(app, 'appStack', {
     // 'seek:system:name': 'TODO: add system name',
   },
 });
+
+// eslint-disable-next-line no-new
+const hookStack = new HookStack(app, 'hookStack');
```

Comment:
This is an interesting one; if we're recommending deploying per-application, it probably needs a unique stack name / function names / etc baked in. I'm unsure on the best approach here!

Reply:
There's normally a `from` function you can use to reference existing resources but not sure that solves where the hooks are deployed

Reply:
Which `function name` are you referring to? Even if we allow `HookStack` to be created with different stack name (it is now hardcoded to [aws-codedeploy-hooks](https://github.com/aurelius0523/aws-codedeploy-hooks/blob/8a9d620b7ff7fbe76112f7c96c06f51e36bb2cb3/packages/infra/src/constructs/stack.ts#L20)), it will still fail if we try to deploy multiple `HookStack`s because lambda names are not unique. I feel like I am missing something here

Also, sorry that I am commenting after work hours, please enjoy your evening! 😄 

![image](https://github.com/user-attachments/assets/3dffb3c7-0d78-4ac9-9c7b-6c1727cac57f)

Reply:
Yeah, those two functions. 

I'm wondering if we could consider merging this as-is (for this thread anyway), with a comment like 

```ts
// TODO: If deploying multiple stacks in one AWS account, deploy HookStack centrally rather than here
```

(Somewhat per @72636c's [comment](https://github.com/seek-oss/skuba/issues/1640#issuecomment-2323854827))

Reply:
I think just purely because it reduces the complexity of the template I'm happy to merge as is

Reply:
@AaronMoat, @samchungy. I have added that comment and also created a changeset. Using changeset for the first time. Let me know if I am not doing that right